src/lib/features/feature-lifecycle/feature-lifecycle-service.ts

```diff
@@ -95,13 +96,20 @@ export class FeatureLifecycleService extends EventEmitter {
         });
         this.eventBus.on(
             CLIENT_METRICS,
-            async (event: ValidatedClientMetrics) => {
-                if (event.environment) {
-                    const features = Object.keys(event.bucket.toggles);
-                    const environment = event.environment;
-                    await this.checkEnabled(() =>
-                        this.featuresReceivedMetrics(features, environment),
-                    );
+            async (events: IClientMetricsEnv[]) => {
+                if (events.length > 0) {
+                    const groupedByEnvironment = groupBy(events, 'environment');
```

Comment:
Since from bulk endpoint we can have events from different environments, we will group it up by environment for lifecycle